Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -393,6 +393,11 @@ class Gx2Fitter {
                     const Logger& /*logger*/) const {
       assert(result.fittedStates && "No MultiTrajectory set");
 
+      if (state.navigation.navigationBreak) {
```

Comment:
what kind of signal are you trying to catch here?

the reason why I ask is that I saw `navigationBreak` being misused for "end of propagation" and I would like to get rid of this signal in the future

Reply:
I want to detect, when we leave our world-volume and stop the actor, when the navigation stopped.

Reply:
I think this coincides with end of propagation so checking for

https://github.com/acts-project/acts/blob/8ccefe18f1263706567ba5848e1ed23ee520fb72/Core/include/Acts/Propagator/Propagator.hpp#L328

https://github.com/acts-project/acts/blob/8ccefe18f1263706567ba5848e1ed23ee520fb72/Core/include/Acts/Propagator/Propagator.hpp#L41

would be better IMO

Reply:
I think I tried to check for this but it didn't work. I can try again.

Reply:
I think I didn't have access to the propagator stage or something like this. Since we also use the '.navigationBreak' in the other fitters I settled with that one. Could we maybe change them later in one go?

Reply:
## Why not use `PropagatorStage`?
It seems the propagator does not know when to finish, therefore we are relying on the navigator. I added the following code for debugging:

```
...
      if (state.navigation.navigationBreak) {
        ACTS_INFO("Actor: finish: state.navigation.navigationBreak");
        std::cout << "navi finished. stage = " << static_cast<int>(state.stage) << std::endl;
        result.finished = true;
      } else {
        std::cout << "not finished. stage = " << static_cast<int>(state.stage) << std::endl;
      }

      if (result.finished) {
        return;
      }
...
```

and got an output similar to this:
```
31754 not finished. stage = 1
31755 not finished. stage = 4
31756 not finished. stage = 4
...
31789 not finished. stage = 4
31790 navi finished. stage = 4
31791 navi finished. stage = 2
```

## How to continue?
As discussed with @andiwand we could keep it with the navigationBreak for now. Andi wants to have a look into it in the future and change it for all fitters.